contracts/FIFOMarket.sol

```diff
@@ -242,6 +283,51 @@ contract FIFOMarket is
     _removal.burnBatch(address(this), batchedIds, batchedAmounts);
   }
 
+  // TODO batch version of this?
+  function reserveRemoval(uint256 removalId) external returns (bool) {
+    address supplierAddress = removalId.supplierAddress();
+    EnumerableSetUpgradeable.UintSet storage supplierSet = _activeSupply[
+      supplierAddress
+    ];
+    require(
+      supplierSet.contains(removalId),
+      "Market: removal not in active supply"
+    );
+    supplierSet.remove(removalId);
```

Comment:
Since [`.remove`](https://docs.openzeppelin.com/contracts/4.x/api/utils#EnumerableSet-remove-struct-EnumerableSet-UintSet-uint256-) returns true if the value was removed (and therefore present) the following should be sufficient:
```suggestion
    require(
      supplierSet.remove(removalId) == true,
      "Market: removal not in active supply"
    );
```